internal/mesh/internal/controllers/routes/xroutemapper/util.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package xroutemapper
+
+import (
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+func DeduplicateRequests(reqs []controller.Request) []controller.Request {
```

Comment:
We can also rely on controller runtime to do that for us as the queue should de-dup all request Ids.

Reply:
I moved it into the mapper test code to help with assertions, but removed it from mainline code as you suggested